contribute/backend/README.md

```diff
@@ -0,0 +1,98 @@
+| /pkg/services/sqlstore | Currently where the database logic resides.                                                                                                                                                                         |
+| /pkg/setting           | Anything related to Grafana global configuration should be dealt with in this package.                                                                                                                              |
+
+## Dependency management
+
+Refer to [UPGRADING_DEPENDENCIES.md](https://github.com/grafana/grafana/blob/main/UPGRADING_DEPENDENCIES.md).
+
+## Ongoing refactoring
+
+These issues are not something we want to address all at once but something we will improve incrementally. Since Grafana is released at a regular schedule the preferred approach is to do this in batches. Not only is it easier to review, but it also reduces the risk of conflicts when cherry-picking fixes from main to release branches. Please try to submit changes that span multiple locations at the end of the release cycle. We prefer to wait until the end because we make fewer patch releases at the end of the release cycle, so there are fewer opportunities for complications.
+
+### Global state
+
+Global state makes testing and debugging software harder and it's something we want to avoid when possible. Unfortunately, there is quite a lot of global state in Grafana.
+
+We want to migrate away from this by using the `inject` package to wire up all dependencies either in `pkg/cmd/grafana-server/main.go` or self-registering using `registry.RegisterService` ex https://github.com/grafana/grafana/blob/main/pkg/services/cleanup/cleanup.go#L25.
+
+### Limit the use of the init() function
+
+Only use the init() function to register services/implementations.
+
+### Settings refactoring
+
+The plan is to move all settings to from package level vars in settings package to the [setting.Cfg](https://github.com/grafana/grafana/blob/df917663e6f358a076ed3daa9b199412e95c11f4/pkg/setting/setting.go#L210) struct. To access the settings, services and components can inject this setting.Cfg struct:
+
+[Cfg struct](https://github.com/grafana/grafana/blob/df917663e6f358a076ed3daa9b199412e95c11f4/pkg/setting/setting.go#L210)
+[Injection example](https://github.com/grafana/grafana/blob/df917663e6f358a076ed3daa9b199412e95c11f4/pkg/services/cleanup/cleanup.go#L20)
+
+### Reduce the use of GoConvey
+
+We want to migrate away from using GoConvey. Instead, we want to use stdlib testing, because it's the most common approach in the Go community and we think it will be easier for new contributors. Read more about how we want to write tests in the [style guide](/contribute/backend/style-guide.md).
+
+### Refactor SqlStore
+
+The `sqlstore` handlers all use a global xorm engine variable. Refactor them to use the `SqlStore` instance.
```

Comment:
can we remove this? I think that we have removed completely the global state since [#48086](https://github.com/grafana/grafana/pull/48086)